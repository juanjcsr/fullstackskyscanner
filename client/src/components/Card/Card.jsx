import React from 'react';
import PropTypes from 'prop-types';

import BpkCard from 'bpk-component-card';
import { BpkGridContainer, BpkGridRow, BpkGridColumn } from 'bpk-component-grid';
import BpkButton from 'bpk-component-button';
import BpkText from 'bpk-component-text';
import { BpkAccordion, BpkAccordionItem, withAccordionItemState } from 'bpk-component-accordion';

import Leg from './../Leg/Leg';
import SegmentList from './../SegmentList/SegmentList';

import STYLES from './Card.scss';


const c = className => STYLES[className] || 'UNKNOWN';

const StatefulAccordionItem = withAccordionItemState(BpkAccordionItem);

const Card = ({
  itinerary, currency, agent, onClick,
}) => {
  const lowestPrice = new Intl.NumberFormat('en-GB', {
    style: 'currency',
    currency: currency.Code,
  }).format(itinerary.PricingOptions[0].Price);
  return (
    <BpkCard onClick={onClick} className={c('Card')}>
      <BpkGridContainer>
        {/* <span id="favicon">{itinerary.icon}</span> */}
        <BpkGridRow>
          <Leg
            legData={itinerary.OutboundLeg}
            directionality={itinerary.OutboundLeg.Directionality}
          />
        </BpkGridRow>
        <BpkGridRow>
          <Leg
            legData={itinerary.InboundLeg}
            directionality={itinerary.InboundLeg.Directionality}
          />
        </BpkGridRow>
        <BpkGridRow>
          <BpkGridColumn width={8}>
            <BpkText tagName="p" bold textStyle="lg">{lowestPrice} </BpkText>
            <BpkText tagName="p">{agent[0].Name}</BpkText>
          </BpkGridColumn>
          <BpkGridColumn width={4}>
            <BpkButton>Select</BpkButton>
          </BpkGridColumn>
        </BpkGridRow>
      </BpkGridContainer>
      {(itinerary.OutboundLeg.Stops.length > 0 || itinerary.InboundLeg.Stops.length > 0) &&
        <BpkAccordion>
          <StatefulAccordionItem id="stops" title="Stops">
            <SegmentList segments={itinerary.OutboundLeg.SegmentsDetail} />
            ----------
            <SegmentList segments={itinerary.InboundLeg.SegmentsDetail} />
          </StatefulAccordionItem>
        </BpkAccordion>
      }

    </BpkCard>
  );
};

Card.propTypes = {
  itinerary: PropTypes.objectOf(PropTypes.any),
  currency: PropTypes.objectOf(PropTypes.any),
  agent: PropTypes.arrayOf(PropTypes.any),
  onClick: PropTypes.func,
};

Card.defaultProps = {
  itinerary: {},
  currency: {},
  agent: [],
  onClick: () => {},
};

export default Card;
