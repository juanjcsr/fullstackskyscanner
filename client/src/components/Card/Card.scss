@import '~bpk-mixins';

.responsive-image__image {
  position: relative;
  width: 100%;
  height: 100%;
}

.Card {
  margin-top: 1em;
}
