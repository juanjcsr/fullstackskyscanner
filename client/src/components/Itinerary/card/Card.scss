@import '~bpk-mixins';

.responsive-image__image {
    position: relative;
    width: 100%;
    height: 100%;
  }