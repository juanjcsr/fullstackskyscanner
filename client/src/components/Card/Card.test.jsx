import React from 'react';
import TestRenderer from 'react-test-renderer';

import Card from './Card';

describe('Card', () => {
  it('should be rendered without data', () => {
    const component = TestRenderer.create(<Card />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly with data', () => {
    const itinerary = {
      OutboundLeg: {
        Id: '11235-1903250720--32302,-32753-1-13554-1903251425',
        SegmentIds: [8, 9],
        OriginStation: 11235,
        DestinationStation: 13554,
        Departure: '2019-03-25T07:20:00',
        Arrival: '2019-03-25T14:25:00',
        Duration: 425,
        JourneyMode: 'Flight',
        Stops: [9889],
        Carriers: [885, 1033],
        OperatingCarriers: [885, 1033],
        Directionality: 'Outbound',
        FlightNumbers: [{ FlightNumber: '681', CarrierId: 885 }, { FlightNumber: '34', CarrierId: 1033 }],
        SegmentsDetail: [{
          OriginStationPlace: [{
            Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
          }],
          DestinationStationPlace: [{
            Id: 9889, ParentId: 876, Code: 'BHD', Type: 'Airport', Name: 'Belfast City',
          }],
          CarrierName: [{
            Id: 885, Code: 'BE', Name: 'Flybe', ImageUrl: 'https://s1.apideeplink.com/images/airlines/BE.png', DisplayCode: 'BE',
          }],
          OperatingCarrier: 885,
          Id: 8,
          OriginStation: 11235,
          DestinationStation: 9889,
          DepartureDateTime: '2019-03-25T07:20:00',
          ArrivalDateTime: '2019-03-25T08:15:00',
          Carrier: 885,
          Duration: 55,
          FlightNumber: '681',
          JourneyMode: 'Flight',
          Directionality: 'Outbound',
        }, {
          OriginStationPlace: [{
            Id: 9889, ParentId: 876, Code: 'BHD', Type: 'Airport', Name: 'Belfast City',
          }],
          DestinationStationPlace: [{
            Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
          }],
          CarrierName: [{
            Id: 1033, Code: 'EI', Name: 'Aer Lingus', ImageUrl: 'https://s1.apideeplink.com/images/airlines/EI.png', DisplayCode: 'EI',
          }],
          OperatingCarrier: 1033,
          Id: 9,
          OriginStation: 9889,
          DestinationStation: 13554,
          DepartureDateTime: '2019-03-25T13:05:00',
          ArrivalDateTime: '2019-03-25T14:25:00',
          Carrier: 1033,
          Duration: 80,
          FlightNumber: '34',
          JourneyMode: 'Flight',
          Directionality: 'Outbound',
        }],
        OriginStationPlace: [{
          Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
        }],
        DestinationStationPlace: [{
          Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
        }],
      },
      InboundLeg: {
        Id: '13554-1903261820--32480-0-11235-1903261945',
        SegmentIds: [10],
        OriginStation: 13554,
        DestinationStation: 11235,
        Departure: '2019-03-26T18:20:00',
        Arrival: '2019-03-26T19:45:00',
        Duration: 85,
        JourneyMode: 'Flight',
        Stops: [],
        Carriers: [881],
        OperatingCarriers: [881],
        Directionality: 'Inbound',
        FlightNumbers: [{ FlightNumber: '1458', CarrierId: 881 }],
        SegmentsDetail: [{
          OriginStationPlace: [{
            Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
          }],
          DestinationStationPlace: [{
            Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
          }],
          CarrierName: [{
            Id: 881, Code: 'BA', Name: 'British Airways', ImageUrl: 'https://s1.apideeplink.com/images/airlines/BA.png', DisplayCode: 'BA',
          }],
          OperatingCarrier: 881,
          Id: 10,
          OriginStation: 13554,
          DestinationStation: 11235,
          DepartureDateTime: '2019-03-26T18:20:00',
          ArrivalDateTime: '2019-03-26T19:45:00',
          Carrier: 881,
          Duration: 85,
          FlightNumber: '1458',
          JourneyMode: 'Flight',
          Directionality: 'Inbound',
        }],
        OriginStationPlace: [{
          Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
        }],
        DestinationStationPlace: [{
          Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
        }],
      },
      OutboundLegId: '11235-1903250720--32302,-32753-1-13554-1903251425',
      InboundLegId: '13554-1903261820--32480-0-11235-1903261945',
      PricingOptions: [{
        Agents: [2365707], QuoteAgeInMinutes: 33, Price: 356.38, DeeplinkUrl: 'http://partners.api.skyscanner.net/apiservices/deeplink/v2?_cje=zOm%2f6FCg6gDOE4rdoXmlE7hM3pD0U74mG5ftPhdP2gYHYNfl2GcrqQa0epbDDGLd&url=https%3a%2f%2fwww.skyscanner.net%2ftransport_deeplink%2f4.0%2fUK%2fen-GB%2fGBP%2febuk%2f2%2f11235.13554.2019-03-25%2c13554.11235.2019-03-26%2fair%2ftrava%2fflights%3fitinerary%3dflight%7c-32302%7c681%7c11235%7c2019-03-25T07%3a20%7c9889%7c2019-03-25T08%3a15%7c55%7cKFLYSS%7cK%7c-%3bflight%7c-32753%7c34%7c9889%7c2019-03-25T13%3a05%7c13554%7c2019-03-25T14%3a25%7c80%7cVBOW26G%7cV%7c-%2cflight%7c-32480%7c1458%7c13554%7c2019-03-26T18%3a20%7c11235%7c2019-03-26T19%3a45%7c85%7cMZ0R%7cM%7c-%26carriers%3d-32302%2c-32753%2c-32480%26operators%3d-32302%3b-32753%2c-32480%26passengers%3d1%26channel%3ddataapi%26cabin_class%3deconomy%26facilitated%3dfalse%26ticket_price%3d356.38%26is_npt%3dfalse%26is_multipart%3dfalse%26client_id%3dskyscanner_b2b%26q_ids%3debuk.11235.4698.190325.190326.1..E%7c8503076537305227810%26commercial_filters%3dfalse%26q_datetime_utc%3d2019-03-19T22%3a54%3a00%26pqid%3dtrue',
      }, {
        Agents: [3503883], QuoteAgeInMinutes: 33, Price: 357.92, DeeplinkUrl: 'http://partners.api.skyscanner.net/apiservices/deeplink/v2?_cje=zOm%2f6FCg6gDOE4rdoXmlE7hM3pD0U74mG5ftPhdP2gYHYNfl2GcrqQa0epbDDGLd&url=https%3a%2f%2fwww.skyscanner.net%2ftransport_deeplink%2f4.0%2fUK%2fen-GB%2fGBP%2fopuk%2f2%2f11235.13554.2019-03-25%2c13554.11235.2019-03-26%2fair%2ftrava%2fflights%3fitinerary%3dflight%7c-32302%7c681%7c11235%7c2019-03-25T07%3a20%7c9889%7c2019-03-25T08%3a15%7c55%7c-%7c-%7c-%3bflight%7c-32753%7c34%7c9889%7c2019-03-25T13%3a05%7c13554%7c2019-03-25T14%3a25%7c80%7c-%7c-%7c-%2cflight%7c-32480%7c1458%7c13554%7c2019-03-26T18%3a20%7c11235%7c2019-03-26T19%3a45%7c85%7c-%7c-%7c-%26carriers%3d-32302%2c-32753%2c-32480%26operators%3d-32302%3b-32753%2c-32480%26passengers%3d1%26channel%3ddataapi%26cabin_class%3deconomy%26facilitated%3dfalse%26ticket_price%3d357.92%26is_npt%3dfalse%26is_multipart%3dfalse%26client_id%3dskyscanner_b2b%26q_ids%3dopuk.11235.13554.190325.190326.1..E%7c-8015069280876441940%26commercial_filters%3dfalse%26q_datetime_utc%3d2019-03-19T22%3a54%3a00%26pqid%3dfalse',
      }, {
        Agents: [2043147], QuoteAgeInMinutes: 9, Price: 361.9, DeeplinkUrl: 'http://partners.api.skyscanner.net/apiservices/deeplink/v2?_cje=zOm%2f6FCg6gDOE4rdoXmlE7hM3pD0U74mG5ftPhdP2gYHYNfl2GcrqQa0epbDDGLd&url=https%3a%2f%2fwww.skyscanner.net%2ftransport_deeplink%2f4.0%2fUK%2fen-GB%2fGBP%2fbfuk%2f2%2f11235.13554.2019-03-25%2c13554.11235.2019-03-26%2fair%2ftrava%2fflights%3fitinerary%3dflight%7c-32302%7c681%7c11235%7c2019-03-25T07%3a20%7c9889%7c2019-03-25T08%3a15%7c55%7cKFLY2S%7cK%7c-%3bflight%7c-32753%7c34%7c9889%7c2019-03-25T13%3a05%7c13554%7c2019-03-25T14%3a25%7c80%7cVB26OWG%7cV%7c-%2cflight%7c-32480%7c1458%7c13554%7c2019-03-26T18%3a20%7c11235%7c2019-03-26T19%3a45%7c85%7cMZ0RO%7cM%7c-%26carriers%3d-32302%2c-32753%2c-32480%26operators%3d-32302%3b-32753%2c-32480%26passengers%3d1%26channel%3ddataapi%26cabin_class%3deconomy%26facilitated%3dfalse%26ticket_price%3d361.90%26is_npt%3dfalse%26is_multipart%3dfalse%26client_id%3dskyscanner_b2b%26q_ids%3dbfuk.11235.13554.190325.190326.1..E%7c8682264064947975486%26commercial_filters%3dfalse%26q_datetime_utc%3d2019-03-19T23%3a18%3a00%26pqid%3dfalse',
      }],
      BookingDetailsLink: { Uri: '/apiservices/pricing/v1.0/b50413b6-3968-46ca-bf13-f8dfaf1db317/booking', Body: 'OutboundLegId=11235-1903250720--32302,-32753-1-13554-1903251425&InboundLegId=13554-1903261820--32480-0-11235-1903261945', Method: 'PUT' },
    };
    const agent = [{"Id":2628363,"Name":"GotoGate","ImageUrl":"https://s1.apideeplink.com/images/websites/gtuk.png","Status":"UpdatesComplete","OptimisedForMobile":true,"Type":"TravelAgent"}];
    const currency = {"Code":"GBP","Symbol":"Â£","ThousandsSeparator":",","DecimalSeparator":".","SymbolOnLeft":true,"SpaceBetweenAmountAndSymbol":false,"RoundingCoefficient":0,"DecimalDigits":2}
    const component = TestRenderer.create(<Card itinerary={itinerary} agent={agent} currency={currency} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
