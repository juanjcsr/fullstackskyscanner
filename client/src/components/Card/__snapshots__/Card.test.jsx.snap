// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be rendered correctly with data 1`] = `
<div
  className="bpk-card bpk-card--padded Card"
  onClick={[Function]}
  role="button"
>
  <div
    className="bpk-grid__container"
  >
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <div>
            <img
              alt=""
              src="https://logos.skyscnr.com/images/airlines/favicon/BE.png"
              style={
                Object {
                  "position": "relative",
                  "width": "30px",
                }
              }
            />
          </div>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            7:20
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            EDI
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <svg
            height="18"
            style={
              Object {
                "height": "1.125rem",
                "width": "1.125rem",
              }
            }
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
            />
          </svg>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            14:25
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            LHR
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            7h 5m 
          </p>
          <p
            className="bpk-text bpk-text--base"
            style={
              Object {
                "color": "rgb(255, 84, 82)",
              }
            }
          >
            1 Stops
          </p>
        </div>
      </div>
    </div>
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <div>
            <img
              alt=""
              src="https://logos.skyscnr.com/images/airlines/favicon/BA.png"
              style={
                Object {
                  "position": "relative",
                  "width": "30px",
                }
              }
            />
          </div>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            18:20
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            LHR
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <svg
            height="18"
            style={
              Object {
                "height": "1.125rem",
                "width": "1.125rem",
              }
            }
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.6 4.5L3.9 9.9c-.4.4-.7.9-.8 1.5-.1.2-.1.4-.1.6 0 .2 0 .4.1.6.1.6.4 1.1.8 1.5l5.7 5.3c.8.8 2.1.7 2.8-.1.8-.8.7-2.1-.1-2.8L9.6 14H19c1.1 0 2-.9 2-2s-.9-2-2-2H9.6l2.7-2.5c.5-.4.7-1 .7-1.5s-.2-1-.5-1.4c-.8-.8-2.1-.8-2.9-.1z"
            />
          </svg>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            19:45
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            EDI
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            1h 25m 
          </p>
          <p
            className="bpk-text bpk-text--base"
            style={
              Object {
                "color": "rgb(0, 215, 117)",
              }
            }
          >
            Direct
          </p>
        </div>
      </div>
    </div>
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div
        className="bpk-grid__column bpk-grid__column--8 bpk-grid__column--padded"
      >
        <p
          className="bpk-text bpk-text--lg bpk-text--bold"
        >
          Â£356.38
           
        </p>
        <p
          className="bpk-text bpk-text--base"
        >
          GotoGate
        </p>
      </div>
      <div
        className="bpk-grid__column bpk-grid__column--4 bpk-grid__column--padded"
      >
        <button
          className="bpk-button"
          disabled={false}
          onClick={null}
          type="button"
        >
          Select
        </button>
      </div>
    </div>
  </div>
  <dl
    className="bpk-accordion"
  >
    <div
      id="stops"
    >
      <dt
        aria-level="3"
        className="bpk-accordion__title"
        role="heading"
      >
        <button
          aria-controls="stops_content"
          aria-expanded={false}
          className="bpk-accordion__toggle-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="bpk-accordion__flex-container"
          >
            <span
              className="bpk-accordion__title-text"
            >
              Stops
            </span>
            <span
              className="bpk-accordion__icon-wrapper"
            >
              <span
                style={
                  Object {
                    "display": "inline-block",
                    "lineHeight": "1.125rem",
                    "marginTop": "0.1875rem",
                    "verticalAlign": "top",
                  }
                }
              >
                <svg
                  className="bpk-accordion__item-expand-icon"
                  height="18"
                  style={
                    Object {
                      "height": "1.125rem",
                      "width": "1.125rem",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 17.5l-7.2-6.4c-.6-.5-.7-1.5-.1-2.1.5-.6 1.5-.7 2.1-.1l5.2 4.6 5.2-4.6c.6-.6 1.6-.5 2.1.1s.5 1.6-.1 2.1L12 17.5z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </button>
      </dt>
      <div
        onTransitionEnd={[Function]}
        style={
          Object {
            "MozTransition": "height 200ms ease ",
            "OTransition": "height 200ms ease ",
            "WebkitTransition": "height 200ms ease ",
            "height": 0,
            "msTransition": "height 200ms ease ",
            "overflow": "hidden",
            "transition": "height 200ms ease ",
          }
        }
      >
        <div>
          <dd
            aria-labelledby="stops_title"
            className="bpk-accordion__content-container"
            id="stops_content"
            role="region"
          >
            <div
              className="bpk-grid__container"
            >
              <div
                className="bpk-grid__row bpk-grid__row--padded"
              >
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <div>
                    <img
                      alt=""
                      src="https://logos.skyscnr.com/images/airlines/favicon/BE.png"
                      style={
                        Object {
                          "position": "relative",
                          "width": "30px",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Departure
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    7:20
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    Edinburgh
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Arrival
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    8:15
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    To:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    Belfast City
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    0h 55m 
                  </p>
                </div>
              </div>
              <div
                className="bpk-grid__row bpk-grid__row--padded"
              >
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <div>
                    <img
                      alt=""
                      src="https://logos.skyscnr.com/images/airlines/favicon/EI.png"
                      style={
                        Object {
                          "position": "relative",
                          "width": "30px",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Departure
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    13:05
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    Belfast City
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Arrival
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    14:25
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    To:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    London Heathrow
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    1h 20m 
                  </p>
                </div>
              </div>
            </div>
            ----------
            <div
              className="bpk-grid__container"
            >
              <div
                className="bpk-grid__row bpk-grid__row--padded"
              >
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <div>
                    <img
                      alt=""
                      src="https://logos.skyscnr.com/images/airlines/favicon/BA.png"
                      style={
                        Object {
                          "position": "relative",
                          "width": "30px",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Departure
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    18:20
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    London Heathrow
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    Arrival
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    19:45
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    To:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    Edinburgh
                  </p>
                </div>
                <div
                  className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
                >
                  <p
                    className="bpk-text bpk-text--sm"
                  >
                    From:
                  </p>
                  <p
                    className="bpk-text bpk-text--sm bpk-text--bold"
                  >
                    1h 25m 
                  </p>
                </div>
              </div>
            </div>
          </dd>
        </div>
      </div>
    </div>
  </dl>
</div>
`;

exports[`Card should be rendered without data 1`] = `
<div
  className="bpk-card bpk-card--padded Card"
  onClick={[Function]}
  role="button"
>
  <div
    className="bpk-grid__container"
  >
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <div>
            <img
              alt=""
              src="https://logos.skyscnr.com/images/airlines/favicon/.png"
              style={
                Object {
                  "position": "relative",
                  "width": "30px",
                }
              }
            />
          </div>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaN:NaN
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <svg
            height="18"
            style={
              Object {
                "height": "1.125rem",
                "width": "1.125rem",
              }
            }
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
            />
          </svg>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaN:NaN
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaNh NaNm 
          </p>
          <p
            className="bpk-text bpk-text--base"
            style={
              Object {
                "color": "rgb(0, 215, 117)",
              }
            }
          >
            Direct
          </p>
        </div>
      </div>
    </div>
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <div>
            <img
              alt=""
              src="https://logos.skyscnr.com/images/airlines/favicon/.png"
              style={
                Object {
                  "position": "relative",
                  "width": "30px",
                }
              }
            />
          </div>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaN:NaN
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
        >
          <svg
            height="18"
            style={
              Object {
                "height": "1.125rem",
                "width": "1.125rem",
              }
            }
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.6 4.5L3.9 9.9c-.4.4-.7.9-.8 1.5-.1.2-.1.4-.1.6 0 .2 0 .4.1.6.1.6.4 1.1.8 1.5l5.7 5.3c.8.8 2.1.7 2.8-.1.8-.8.7-2.1-.1-2.8L9.6 14H19c1.1 0 2-.9 2-2s-.9-2-2-2H9.6l2.7-2.5c.5-.4.7-1 .7-1.5s-.2-1-.5-1.4c-.8-.8-2.1-.8-2.9-.1z"
            />
          </svg>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaN:NaN
          </p>
          <p
            className="bpk-text bpk-text--base"
          >
            
          </p>
        </div>
        <div
          className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
        >
          <p
            className="bpk-text bpk-text--base"
          >
            NaNh NaNm 
          </p>
          <p
            className="bpk-text bpk-text--base"
            style={
              Object {
                "color": "rgb(0, 215, 117)",
              }
            }
          >
            Direct
          </p>
        </div>
      </div>
    </div>
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <div
        className="bpk-grid__column bpk-grid__column--8 bpk-grid__column--padded"
      >
        <p
          className="bpk-text bpk-text--lg bpk-text--bold"
        >
          Â£0.00
           
        </p>
        <p
          className="bpk-text bpk-text--base"
        >
          
        </p>
      </div>
      <div
        className="bpk-grid__column bpk-grid__column--4 bpk-grid__column--padded"
      >
        <button
          className="bpk-button"
          disabled={false}
          onClick={null}
          type="button"
        >
          Select
        </button>
      </div>
    </div>
  </div>
</div>
`;
